lighthouse-core/runner.js

```diff
@@ -153,6 +153,52 @@ class Runner {
         .readdirSync(path.join(__dirname, './gather/gatherers'))
         .filter(f => /\.js$/.test(f));
   }
+
+  /**
+   * Resolves the location of the specified plugin and returns an absolute
+   * string path to the file. Used for loading custom audits and gatherers.
+   * Throws an error if no plugin is found.
+   * @param {string} plugin
+   * @param {string=} configDir The absolute path to the directory of the config file, if there is one.
+   * @param {string=} category Optional plugin category (e.g. 'audit') for better error messages.
+   * @return {string}
+   * @throws {Error}
+   */
+  static resolvePlugin(plugin, configDir, category) {
+    // First try straight `require()`. Unlikely to be specified relative to this
+    // file, but adds support for Lighthouse plugins in npm modules as
+    // `require()` walks up parent directories looking inside any node_modules/
+    // present. Also handles absolute paths.
+    try {
+      return require.resolve(plugin);
+    } catch (e) {}
+
+    // See if the plugin resolves relative to the current working directory.
+    // Most useful to handle the case of invoking Lighthouse as a module, since
+    // then the config is an object and so has no path.
+    const cwdPath = path.resolve(process.cwd(), plugin);
+    try {
+      return require.resolve(cwdPath);
+    } catch (e) {}
+
+    const errorString = 'Unable to locate ' +
```

Comment:
it has to be split up no matter what due to the line length restriction. I tried a couple of ways to do this (like ``category = category ? `${category}: ` : ''`` ahead of time, and then using `${category}` inside the template string), but this seemed like a clearer way to see what it's doing with `category`.